packages/perseus-editor/src/widgets/image-editor/components/decorative-toggle.test.tsx

```diff
@@ -0,0 +1,194 @@
+        // Arrange
+        render(
+            <DecorativeToggle
+                hasPopulatedFields={false}
+                onChange={mockOnChange}
+            />,
+            {wrapper: RenderStateRoot},
+        );
+
+        // Act
+        const toggle = screen.getByRole("switch", {name: "Decorative"});
+        await userEvent.click(toggle);
+
+        // Assert
+        expect(mockOnChange).toHaveBeenCalledWith({decorative: true});
+    });
+
+    test("calls onChange with decorative: false when toggle is clicked and decorative is true", async () => {
+        // Arrange
+        render(<DecorativeToggle decorative={true} onChange={mockOnChange} />, {
+            wrapper: RenderStateRoot,
+        });
+
+        // Act
+        const toggle = screen.getByRole("switch", {name: "Decorative"});
+        await userEvent.click(toggle);
+
+        // Assert
+        expect(mockOnChange).toHaveBeenCalledWith({decorative: false});
+    });
+
+    test("shows warning dialog when toggle is clicked and fields are populated", async () => {
+        // Arrange
+        const mockConfirm = jest.spyOn(window, "confirm").mockReturnValue(true);
+        render(
+            <DecorativeToggle
+                hasPopulatedFields={true}
+                onChange={mockOnChange}
+            />,
+            {wrapper: RenderStateRoot},
+        );
+
+        // Act
+        const toggle = screen.getByRole("switch", {name: "Decorative"});
+        await userEvent.click(toggle);
+
+        // Assert
+        expect(mockConfirm).toHaveBeenCalledWith(
+            "Setting this image as decorative will automatically reset all other fields (title, caption, alt text, and long description). Do you want to continue?",
+        );
+        expect(mockOnChange).toHaveBeenCalledWith({
+            decorative: true,
+            alt: undefined,
+            caption: undefined,
+            title: undefined,
+            longDescription: undefined,
+        });
+    });
+
+    test("does not call onChange when user cancels the warning dialog", async () => {
+        // Arrange
+        const mockConfirm = jest
+            .spyOn(window, "confirm")
+            .mockReturnValue(false);
+        render(
+            <DecorativeToggle
+                hasPopulatedFields={true}
+                onChange={mockOnChange}
+            />,
+            {wrapper: RenderStateRoot},
+        );
+
+        // Act
+        const toggle = screen.getByRole("switch", {name: "Decorative"});
+        await userEvent.click(toggle);
+
+        // Assert
+        expect(mockConfirm).toHaveBeenCalledWith(
+            "Setting this image as decorative will automatically reset all other fields (title, caption, alt text, and long description). Do you want to continue?",
+        );
+        expect(mockOnChange).not.toHaveBeenCalled();
+    });
+
+    test("does not show warning dialog when fields are not populated", async () => {
```

Comment:
Is this the same as the test "calls onChange with decorative: true when toggle is clicked and no fields are populated"?